toytools/datasets/simple_toyzero.py

```diff
@@ -0,0 +1,150 @@
+    """Simple Dataset to load toyzero images on demand with no preprocessing.
+
+    Parameters
+    ----------
+    path : str
+        Path where the toyzero dataset is located.
+    crop_shape : (int, int) or None
+        If specified, then crop images down to shape specified by `crop_shape`.
+        Default: None
+    expansion_factor : int
+        Number of cropped regions to select from a single image.
+        Default: 1
+    is_train : bool
+        `is_train` flag controls overall behavior of the
+        `SimpleToyzeroDataset`. If `is_train` is True, then the cropped regions
+        of toyzero images are selected at random. Otherwise, the cropped
+        regions are selected in a repdorucible manner using `seed` parameter.
+        This flag also controls which part of the entire toyzero dataset this
+        object will focus on.
+        Default: False
+    planes : Set[str] or None
+        A set of planes to select. If None, then all planes are selected.
+        Default: None
+    seed : int
+        Seed used to initialize random number generators.
+        Default: 0
+    transform : Callable or None,
+        Optional transformation to apply to images.
+        E.g. torchvision.transforms.RandomCrop.
+        Default: None
+    val_size : float or int
+        Fraction of the toyzero dataset that will be used as a validation
+        sample. If val_size <= 1, then it is treated as a fraction, i.e.
+        (size of val sample) = `val_size` * (size of toyzero dataset)
+        Otherwise, (size of val sample) = `val_size`.
+        Default: 0.2
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        path,
+        crop_shape       = None,
+        expansion_factor = 1,
+        is_train         = False,
+        planes           = None,
+        seed             = 0,
+        transform        = None,
+        val_size         = 0.2,
+    ):
+        super().__init__(path)
+
+        self._is_train = is_train
+        self._images   = []
+
+        self._indices   = []
+        self._len       = 0
+
+        self._crop_shape = crop_shape
+        self._transform  = transform
+        self._seed       = seed
+        self._efactor    = expansion_factor
+        self._prg        = np.random.default_rng(seed)
+        self._planes     = planes
+        self._val_size   = val_size
+
+        self._collect_images()
+        self._split_dataset()
+
+    def _collect_images(self):
+        """Locate all matching images in the dataset directory."""
+        images = collect_toyzero_images(self._path)
+        self._images = filter_images(images, None, self._planes)
+
+    def _split_dataset(self):
+        """Split images into training/validation samples."""
+        indices = np.arange(len(self._images))
+        self._prg.shuffle(indices)
+
+        val_size = self._val_size
+
+        if val_size <= 1:
+            val_size = int(len(indices) * val_size)
```

Comment:
I might be misunderstanding something, but why, when the input `val_size` is a fraction, the integer `val_size` doesn't have to divide `self._efactor`, but when `val_size` is an integer, it does?

Reply:
Yes, sorry, this is counterintuitive behavior, and probably I should remove it. The motivation was, that `efactor` indicates how many cropped regions to sample from a single image. So, if we want our validation dataset to have 1000 cropped regions, then first we need to have `1000 / efactor` full images. And then sample `efactor` of cropped regions from each full image.

On the other hand, if we want our validation dataset to be 20% of the entire dataset, then we can simply select 20% of all full images and sample `efactor` of cropped regions from each.

Reply:
Hi Dmitrii, thank you for the explanation! I thought this must be the case when I read it yesterday, but it is, as you said, a little counterintuitive because `val_size` seems to mean two slightly different things here. I think maybe the explanation your give here is sufficient and if we want to keep the code.